packages/__tests__/plugin-conventions/name-convention.spec.ts

```
@@ -0,0 +1,33 @@
+import { nameConvention } from '@aurelia/plugin-conventions';
+import { assert } from '@aurelia/testing';
+
+describe('nameConvention', function () {
+  it('complains about empty input', function() {
+    assert.throws(() => nameConvention(''));
```

maybe throw on anything that is potentially invalid? Just to keep the spirit of our testing, started by @fkleuver

---

It's kind of internal API to translate ClassName. BTW, what kind of invalid input you has in mind?

Even `WhatEverName` will be a custom element based on convention.

---

Undefined, false symbol null number

---

I thought because this is TS project, you don't need to do that kind of tests anymore?

---

I think the compiler will reject that kind of test code to build.

---

@fkleuver I tested `nameConvention(null)`, the `npm run dev` didn't complain about the usage against the typing `nameConvention(className: string)` in the test code. Is there any missed typing setup in our test code compilation?